controller/src/test/java/io/pravega/controller/task/Stream/StreamMetadataTasksTest.java

```diff
@@ -458,6 +461,294 @@ public void retentionStreamTest() throws Exception {
+
+        streamStorePartialMock.createStream(SCOPE, streamName, configuration, System.currentTimeMillis(), null, executor).get();
+        streamStorePartialMock.setState(SCOPE, streamName, State.ACTIVE, null, executor).get();
+
+        assertNotEquals(0, consumer.getCurrentSegments(SCOPE, streamName).get().size());
+        WriterMock requestEventWriter = new WriterMock(streamMetadataTasks, executor);
+        streamMetadataTasks.setRequestEventWriter(requestEventWriter);
+
+        // region size based retention on stream cuts on epoch 0
+        // region no previous streamcut
+        // first retention iteration
+        // streamcut1: 19 bytes(0/9,1/10)
+        long recordingTime1 = System.currentTimeMillis();
+        Map<Integer, Long> map1 = new HashMap<>();
+        map1.put(0, 9L);
+        map1.put(1, 10L);
+        long size = streamStorePartialMock.getSizeTillStreamCut(SCOPE, streamName, map1, null, executor).join();
+        assertEquals(size, 19);
+        StreamCutRecord streamCut1 = new StreamCutRecord(recordingTime1, size, map1);
+
+        doReturn(CompletableFuture.completedFuture(streamCut1)).when(streamMetadataTasks).generateStreamCut(
+                anyString(), anyString(), any());
+
+        streamMetadataTasks.retention(SCOPE, streamName, retentionPolicy, recordingTime1, null).get();
+        // verify that one streamCut is generated and added.
+
+        List<StreamCutRecord> list = streamStorePartialMock.getStreamCutsFromRetentionSet(SCOPE, streamName, null, executor).get();
+        assertTrue(list.contains(streamCut1));
+        // endregion
+
+        // region stream cut exists but latest - previous < retention.size
+        // second retention iteration
+        // streamcut2: 100 bytes(0/50, 1/50)
+        Map<Integer, Long> map2 = new HashMap<>();
+        map2.put(0, 50L);
+        map2.put(1, 50L);
+        long recordingTime2 = recordingTime1 + Duration.ofMinutes(Config.MINIMUM_RETENTION_FREQUENCY_IN_MINUTES).toMillis() + 1;
+        size = streamStorePartialMock.getSizeTillStreamCut(SCOPE, streamName, map2, null, executor).join();
+        assertEquals(size, 100L);
+        StreamCutRecord streamCut2 = new StreamCutRecord(recordingTime2, size, map2);
+        doReturn(CompletableFuture.completedFuture(streamCut2)).when(streamMetadataTasks).generateStreamCut(
+                anyString(), anyString(), any());
+
+        streamMetadataTasks.retention(SCOPE, streamName, retentionPolicy, recordingTime2, null).get();
+        list = streamStorePartialMock.getStreamCutsFromRetentionSet(SCOPE, streamName, null, executor).get();
+        StreamProperty<StreamTruncationRecord> truncProp = streamStorePartialMock.getTruncationProperty(SCOPE, streamName, true, null, executor).get();
+        // verify that only one stream cut is in retention set. streamCut2 is not added
+        // verify that truncation did not happen
+        assertTrue(list.contains(streamCut1));
+        assertTrue(list.contains(streamCut2));
```

Comment:
This comment above:

```
// verify that only one stream cut is in retention set. streamCut2 is not added
```

implies that this assertion should actually be false.

Reply:
copy paste..  its the comment that is wrong. fixed it